### ISSUE TITLE
API key needed now

### ISSUE THREAD

**Issue by stevehoek** [yenoiwesa/homebridge-connexoon#94]
I noticed this morning that Tahoma has released a new iOS app and deprecated their old one.

Then later today I noticed in Homebridge that there are lots of errors from this plugin saying that an API key is needed for all requests.

Is this plugin still maintained such that this could be added?

**Comment by mdiberardino-atlassian** [comment]
Hey @stevehoek I am not actively maintaining this repository anymore because I moved on to a fully local version of the Somfy controls that use RF to the blinds from a Raspberry Pi.

But if you are able to find me a replacement API key I am happy to swap it out in code and release a new version of the plugin.

Maybe https://github.com/dubocr/homebridge-tahoma has been updated already and it can be found there?

**Comment by stevehoek** [comment]
@mdiberardino-atlassian thanks for the reply.  The homebridge-tahoma plugin hasnt been updated either, and no one has mentioned it as an issue but it just happened today so its early.

I guess I figured Somfy/Tahoma was moving to per-user API keys but it could just be a general API key.  I found some info about activating Developer Mode on my account and then generating an API key via this page, but it didnt work.

https://dev.duboc.pro/overkiz
(not even sure if this is official, but it was linked to from the somfysystems.com "My Automations" page where you can activate developer mode)

**Comment by mdiberardino-atlassian** [comment]
> I guess I figured Somfy/Tahoma was moving to per-user API keys

Yes I was afraid of that, to be honest that leaked API key for app authentication coupled with username/password was a bit suspicious to begin with. They might have deprecated that authentication method.

The request headers might have changed too to accept that developer mode API key, mmhh.

**Comment by mdiberardino-atlassian** [comment]
@stevehoek did the issue eventually resolve itself or is the plugin broken for good now?

If it is, I will put a deprecation notice in the README file to let people know.

**Comment by Lammiwinks** [comment]
This works for me today still. So definitly still working.
Whats the local version you are using @mdiberardino-atlassian? I'd love to use the local connection rather than the cloud one with this app (like you can with the Tahoma version). Cheers!

**Comment by yenoiwesa** [comment]
Hey @Lammiwinks this is the plugin I used to use to have local network interfacing with the RTS blinds and custom RF module: https://github.com/yenoiwesa/homebridge-somfy-pi

However I moved home and for my new one I bought the Somfy Tahoma Switch Hub, and will buy Zigbee Sonesse blinds. The Somfy Tahoma Switch hub is Matter compatible and offers local control without the need for the Somfy servers connection. It is fully homekit native thanks to Matter, so I guess there is no longer a reason to use a homebridge plugin for these integrations.

**Comment by Lammiwinks** [comment]
Thank you for sharing! Really appreciate your help. And makes total sense to move to a more native solution.

I’m in a 2 year old house with 20+ somfy RTS blinds that the old owner put in. So your plugin has been a lifesaver vs running around with 5+ remotes!

I have also considered going down the Bond Path now that it too natively supports HomeKit.

**Comment by drekka** [comment]
Hello everyone, my home bridge setup running 5 blinds has been fine for years using this plugin, then last week it stopped working. The logs full of `error: 'An API key is required to access this setup` messages. 

I presume from the discussion here that this plugin will no longer work, although I don't understand why it's taken this long to fail. 

My question is do I need to enable developer mode on the Somfy account, grab an API key and switch to the https://github.com/dubocr/homebridge-tahoma plugin to get the blinds working again? Or am I miss-reading things :-)
